### ISSUE TITLE
Configure screen menu no longer changes icon colours etc. to match profile

### ISSUE THREAD

**Issue by mikehgentry** [osmandapp/OsmAnd#18091]
### Description

The colours only update if you select one of the widget panels then hit the back arrow, or exit and re-enter the menu.

### How to reproduce

Menu -> Configure screen

Change profile.

### Actual result

Icons remain in the initial profile's colour scheme

### Expected result

Icons change to match the newly selected profile

**Environment**
OsmAnd Version: + 4.6.0
Android/iOS version: 11
Device model: XCover 4s

**Comment by DmitryAlexei** [comment]
**Steps to reproduce:**
- in one profile (for example *Browse map*) set *Light app theme* (go to Menu > Settings > Browse map > General settings > light)
- in other profile (for example *Driving*) set *Dark App theme* (go to Menu > Settings > Browse map > General settings > dark)
- Go to Menu > Configure screen 
- Switch between *Browse map* and *Driving* profiles, go to the *Left* or *Right panel* in driving profile

**Actual result:**
- When switching between *Browse map* and *Driving* ***App theme* doesn't change**
- When you go to the *Left* or *Right panel* color scheme changes
- For comparison added recording from *Configure map menu*

<table style="width:100%">
  <tr>
    <th>Issue Configure screen</th>
    <th>Configure map for comparison</th>
  </tr>
  <tr>
    <td><video source src="https://github.com/osmandapp/OsmAnd/assets/116079541/75b5b950-5d51-4c41-ab8e-23c899d0eb0c"  width="200" height="auto"/></td>
    <td><video source src="https://github.com/osmandapp/OsmAnd/assets/116079541/7b52b799-0b98-451f-97cd-9d3cc6e6abc2"  width="200" height="auto"/></td>
  </tr>

**Comment by mikehgentry** [comment]
I meant the icon colours selected in Configure profile -> Profile appearance, but that may be the same bug

**Comment by DmitryAlexei** [comment]
Indeed, the profile color also does not change

https://github.com/osmandapp/OsmAnd/assets/116079541/517dd9a7-349a-4721-b3e6-652a197e13b8

**Comment by mikehgentry** [comment]
Probably more seriously, the status of the other options at the bottom of this menu ('Distance by tap' and everything) doesn't update on profile switch.

**Comment by DmitryAlexei** [comment]
Yes, there are errors in displaying the current state of *Distance by tap* option and *Quick action* widget
OsmAnd~ 4.6.0#806m, released: 2023-09-25

**Steps to reproduce:**
- activate Distance by tap (or quick action widget) in one profile and disable in another
- switch between profiles and see if status of Distance by tap option or Quick action widget are enabled or disabled

**Actual result:**
- status of Distance by tap and Quick action follows last used profile
- when you tap on one of this option, you can see its real status

<table style="width:100%">
  <tr>
    <th>Quick action button </th>
    <th>Distance by tap option </th>
  </tr>
  <tr>
    <td><video source src="https://github.com/osmandapp/OsmAnd/assets/116079541/9113b370-9b19-4bca-9dc6-c9f04b63a9e1"  width="200" height="auto"/></td>
    <td><video source src="https://github.com/osmandapp/OsmAnd/assets/116079541/87bfcec7-52a0-42d6-84ab-7871673518ed"  width="200" height="auto"/></td>
  </tr>

**Comment by mikehgentry** [comment]
I'm seeing the same for Compass and 3D mode.

**Comment by mikehgentry** [comment]
This (at least the switch state part of it, rather than the cosmetic part) seems like a significant UI bug. Should it be given a milestone?

**Comment by vshcherb** [comment]
To do https://github.com/osmandapp/OsmAnd/issues/18091#issuecomment-1737368780
